optuna/pruners/threshold.py

```diff
@@ -0,0 +1,59 @@
+from optuna.pruners import BasePruner
+
+from optuna.type_checking import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from optuna.structs import FrozenTrial
+    from optuna.study import Study
+
+
+class ThresholdPruner(BasePruner):
+    """Pruner to detect abnormal metrics of the trials.
+
+    Prune if a metric exceeds upper bound threshold or
+    falls behind lower bound threshold which users specify.
+
+    Example:
+
+        .. code::
+
+            >>> from optuna import create_study
+            >>> from optuna.pruners import ThresholdPruner
+            >>>
+            >>> study = create_study(pruner=ThresholdPruner(upper=500))
+            >>> study.optimize(objective)
+
+    Args
+        lower_bound:
+            minimum value which determines whether pruner prunes or not
+            (If value is smaller than lower_bound, it prunes)
+        upper_bound:
+            maximum value which determines whether pruner prunes or not
+            (If value is larger than upper_bound, it prunes)
+
+    """
+
+    def __init__(self, lower_bound=None, upper_bound=None):
```

Comment:
Could be a matter of taste, but what do you think of shorter names? I personally prefer these as they still communicate the intent well enough.
```suggestion
    def __init__(self, lower=None, upper=None):
```